### PR TITLE
Release maintenance/flocker 1.0.1/disable restart policies floc 2467

### PR DESCRIPTION
**Pull request by exarkun**
Fixes https://clusterhq.atlassian.net/browse/FLOC-2467
Update of https://github.com/ClusterHQ/flocker/pull/1696 (changing branch name ... which matters ...)

### CODE REVIEW

File: flocker/node/test/test_deploy.py [hunk 1]
```diff
@@ -1994,6 +1997,52 @@ def test_missing_volume(self):
...
+        state = NodeState(
+            uuid=uuid4(), hostname=u"192.0.2.11",
+            applications={app_state}, used_ports=[],
+        )
+        config = to_node(state).set(applications={app_config})
+        assert_application_calculated_changes(
+            self, state, config, set(),
+            restart(app_state, app_config, state),
+        )
+
```
Comment (itamarst) [inline flocker/node/test/test_deploy.py]: Add (a couple of?) tests that if state is `RestartNever` there is no restart, regardless of config restart policy.

Comment (itamarst) [inline flocker/node/test/test_deploy.py]: The worry is that if policy is e.g. RestartAlways one will get infinite restarts.

**Comment by itamarst** [comment]
Looks good if new tests don't turn up a bug. If they do resubmit, otherwsie merge when buildbot is hapy.

**Comment by exarkun** [comment]
@itamarst Can't resubmit since the state is "Accepted".  The tests did turn up another bug though.

**Comment by itamarst** [comment]
You can can click Merged and then resurrect and then clickclikclickickclckclck but sure. Looks good to me, merge if buildbot is happy (haha).

**Pull request merged**